File: react-native-pytorch-core/ios/CanvasManager.m
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_REMAP_MODULE(PyTorchCoreCanvasView, CanvasManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onContext2D, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(width, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(height, NSNumber)
RCT_EXTERN_METHOD(requiresMainQueueSetup)
@end

File: react-native-pytorch-core/ios/CanvasManager.swift
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation

@objc(CanvasManager)
class CanvasManager: RCTViewManager {

    var canvas = DrawingCanvasView()

    override func view() -> UIView! {
        canvas.backgroundColor = .white
        return canvas
    }

    override static func requiresMainQueueSetup() -> Bool {
       return true
     }
}
